File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    # stream_from 'chat_channel'
    stream_from "chat"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def get_user_data
    puts "Gets here"
    data = {
      id: current_user.id,
      email: current_user.email,
      username: current_user.email.split('@')[0],
    }
    ActionCable.server.broadcast 'chat_channel', data
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("ChatChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log("ChatChannel connected")
    this.perform("get_user_data")
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
    console.log("ChatChannel disconnected")
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    console.log("ChatChannel received:", data.email )
  }
});
